packages/hooks/src/data/QueryData.ts

```diff
@@ -118,6 +115,85 @@ export class QueryData<TData, TVariables> extends OperationData {
     delete this.previousData.result;
   }
 
+  public getOptions() {
+    const options = super.getOptions();
+    const lazyOptions = this.lazyOptions || {};
+    const updatedOptions = {
+      ...options,
+      variables: {
+        ...options.variables,
+        ...lazyOptions.variables
+      },
+      context: {
+        ...options.context,
+        ...lazyOptions.context
+      }
+    };
+
+    // Triggering a query in lazy mode overrides `skip` settings.
+    if (options.lazy) {
+      updatedOptions.skip = false;
+    }
+
+    return updatedOptions;
+  }
+
+  private initialLazyResult(): QueryTuple<TData, TVariables> {
+    return [
+      {
+        loading: false,
+        networkStatus: NetworkStatus.ready,
+        called: false,
+        data: undefined
+      } as QueryResult<TData, TVariables>,
+      this.executeLazy.bind(this)
+    ];
+  }
+
+  private executeLazy(options?: QueryLazyOptions<TVariables>) {
+    this.runLazy = true;
+    this.lazyOptions = options;
+    this.forceUpdate();
+  }
+
+  private getExecuteResult(): TLazy extends undefined
+    ? QueryResult<TData, TVariables>
+    : QueryTuple<TData, TVariables> {
+    const result = this.getQueryResult();
+    this.startQuerySubscription();
+    return (this.getOptions().lazy !== undefined
+      ? [result, this.executeLazy.bind(this)]
+      : result) as any;
+  }
+
+  private getExecuteSsrResult(): TLazy extends undefined
+    ? QueryResult<TData, TVariables>
+    : QueryTuple<TData, TVariables> {
+    let result;
+
+    const ssrLoading = {
+      loading: true,
+      networkStatus: NetworkStatus.loading,
+      called: true,
+      data: {}
+    };
+
+    if (this.context && this.context.renderPromises) {
+      result = this.context.renderPromises.addQueryPromise(
+        this,
+        this.getExecuteResult.bind(this)
```

Comment:
Let’s avoid binding in favor of using the class property + arrow function trick, since rebinding functions is expensive, and arrow functions are faster in some engines than bound functions.

Reply:
I didn't realize rebinding was more expensive than using arrow functions, so TIL. That being said, I still visually prefer the explicitness of seeing `.bind` 😢. But, performance trumps my preference, so done!